src/Controller/Api/ThemesController.php

```diff
@@ -2,38 +2,24 @@
 
 namespace App\Controller\Api;
 
+use App\Repository\ThemeRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\Routing\Attribute\Route;
 
 #[Route('/api')]
 final class ThemesController extends AbstractController
 {
+    private $themeRepository;
+
+    public function __construct(ThemeRepository $themeRepository)
+    {
+        $this->themeRepository = $themeRepository;
+    }
+
     #[Route('/themes', name: 'app_themes', methods: ['GET'])]
     public function index(): JsonResponse
     {
-        return $this->json([
-            'themes' => [
-                [
-                    'code' => 'environment',
-                    'id' => 1234,
-                    'parentId' => null,
-                    'children' => [
-                        [
-                            'code' => 'some_sub_theme',
-                            'id' => 1024,
-                            'parentId' => 1234,
-                            'children' => [
-                                [
-                                    'code' => 'yet_another_sub_theme',
-                                    'id' => 2048,
-                                    'parentId' => 1024,
-                                ],
-                            ],
-                        ],
-                    ],
-                ],
-            ],
-        ]);
+        return $this->json(['themes' => $this->themeRepository->findAllHierarchical()]);
```

Comment:
@magrigsdev there is probably a better way, but this will suffice for now I think.